GVFS/GVFS.FunctionalTests/Tests/EnlistmentPerFixture/IncludedFolderTests.cs

```diff
@@ -0,0 +1,262 @@
+
+            this.gvfsProcess.RemoveIncludedFolders(siblingPath);
+            folder.ShouldNotExistOnDisk(this.fileSystem);
+            folder = this.Enlistment.GetVirtualPathTo(this.mainIncludedFolder);
+            folder.ShouldBeADirectory(this.fileSystem);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+        }
+
+        [TestCase, Order(4)]
+        public void AddingSubfolderShouldKeepParentRecursive()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            // Add subfolder of main folder and make sure it stays recursive
+            string subFolder = Path.Combine(this.mainIncludedFolder, "Properties");
+            this.gvfsProcess.AddIncludedFolders(subFolder);
+            string folder = this.Enlistment.GetVirtualPathTo(subFolder);
+            folder.ShouldBeADirectory(this.fileSystem);
+            this.ValidateIncludedFolders(this.mainIncludedFolder, subFolder);
+
+            folder = this.Enlistment.GetVirtualPathTo(this.mainIncludedFolder, "CommandLine");
+            folder.ShouldBeADirectory(this.fileSystem);
+        }
+
+        [TestCase, Order(5)]
+        [Category(Categories.WindowsOnly)]
+        public void CreatingFolderShouldAddToIncludedListAndStartProjecting()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string newFolderPath = Path.Combine(this.Enlistment.RepoRoot, "GVFS", "GVFS.Common");
+            newFolderPath.ShouldNotExistOnDisk(this.fileSystem);
+            Directory.CreateDirectory(newFolderPath);
+            newFolderPath.ShouldBeADirectory(this.fileSystem);
+            string[] fileSystemEntries = Directory.GetFileSystemEntries(newFolderPath);
+            fileSystemEntries.Length.ShouldEqual(32);
+
+            string projectedFolder = Path.Combine(newFolderPath, "Git");
+            projectedFolder.ShouldBeADirectory(this.fileSystem);
+            fileSystemEntries = Directory.GetFileSystemEntries(projectedFolder);
+            fileSystemEntries.Length.ShouldEqual(13);
+
+            string projectedFile = Path.Combine(newFolderPath, "ReturnCode.cs");
+            projectedFile.ShouldBeAFile(this.fileSystem);
+        }
+
+        [TestCase, Order(5)]
+        [Category(Categories.MacOnly)]
+        public void CreateFolderThenFileShouldAddToIncludedListAndStartProjecting()
+        {
+            this.gvfsProcess.AddIncludedFolders(this.mainIncludedFolder);
+            this.ValidateIncludedFolders(this.mainIncludedFolder);
+
+            string newFolderPath = Path.Combine(this.Enlistment.RepoRoot, "GVFS", "GVFS.Common");
+            newFolderPath.ShouldNotExistOnDisk(this.fileSystem);
+            Directory.CreateDirectory(newFolderPath);
+            string newFilePath = Path.Combine(newFolderPath, "test.txt");

```

Comment:
Could you also add a test that does this on Mac but adds a file that is already in the projection?